src/poketch_data.c

```diff
@@ -0,0 +1,360 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "pokemon.h"
+#include "struct_decls/struct_021C0794_decl.h"
+
+#include "unk_020244AC.h"
+#include "poketch_data.h"
+#include "pokemon_icon.h"
+#include "savedata/save_table.h"
+
+#define POKETCH_REGISTRY_SIZE 32
+
+/**
+ * @brief All of the Poketch's internally tracked data including settings, registered apps, and the data for some apps (eg. pedometer, alarm clock).
+ */
+typedef struct PoketchData {
+    u8 unk_00_0 : 1;
+    u8 pedometerEnabled : 1;    //!< Whether or not the pedometer is registered and will take step count updates.
+    u8 unk_00_2 : 1;
+    u8 screenColor : 3;         //!< Screen palette color (valid values defined in poketch_data.h)
+
+    u8 unk_00_6 : 2;            //!< unused
+
+    s8 appCount;                                //!< Number of currently registered apps
+    s8 appIndex;                                //!< Currently selected app
+    u8 appRegistry[POKETCH_REGISTRY_SIZE];      //!< Registration status of all apps. Indices 0-24 correspond to the App IDs in poketch_data.h. Indices 25-31 are unused.
+    
+    u32 pedometer;         //!< Step counter
+
+    u16 alarmSet : 1;      //!< Whether or not the alarm is currently enabled.
+    u16 alarmHour : 5;     //!< Current Hour setting on the alarm clock.
+    u16 alarmMinute : 6;   //!< Current Minute setting on the alarm clock.
+    
+    u16 unk_28_12 : 4;     //!< unused
+
+    u8 unk_2A[120];
+    u32 unk_A4;
+    u8 unk_A8;
+
+    struct {
+        u8 x;                                       //!< X coordinate of map marker
+        u8 y;                                       //!< Y coordinate of map marker
+    } markMapPositions[POKETCH_MAPMARKER_COUNT];    //!< Map markers
+
+    struct {
+        u16 unk_00;
+        u16 unk_02;
+        u32 unk_04;
+    } unk_B8[12];
+} PoketchData;
+
+
+int Poketch_SaveSize (void)
```

Comment:
Convention in the repo is no space between function label and the first open-paren, at least once the function is labeled.

```suggestion
int Poketch_SaveSize(void)
```